app/forms/api/profile_creation_form.rb

```diff
@@ -0,0 +1,162 @@
+      create_gpo_entry if session[:address_verification_mechanism] == 'gpo'
+    end
+
+    def phone_confirmed?
+      session[:vendor_phone_confirmation] == true && session[:user_phone_confirmation] == true
+    end
+
+    def complete_profile
+      current_user.pending_profile&.activate
+      move_pii_to_user_session
+    end
+
+    def move_pii_to_user_session
+      return if session[:decrypted_pii].blank?
+      user_session[:decrypted_pii] = session.delete(:decrypted_pii)
+    end
+
+    def create_gpo_entry
+      move_pii_to_user_session
+      if session[:pii].is_a?(String)
+        session[:pii] = Pii::Attributes.new_from_json(user_session[:decrypted_pii])
+      end
+      confirmation_maker = GpoConfirmationMaker.new(
+        pii: session[:pii], service_provider: service_provider,
+        profile: profile
+      )
+      confirmation_maker.perform
+
+      @gpo_otp = confirmation_maker.otp
+    end
+
+    def build_profile_maker
+      Idv::ProfileMaker.new(
+        applicant: user_bundle_payload,
+        user: current_user,
+        user_password: user_password,
+      )
+    end
+
+    def current_user
+      return nil unless user_bundle_headers
+      return @current_user if defined?(@current_user)
+      @current_user = User.find_by(uuid: user_bundle_headers['sub'])
+    end
+
+    def set_idv_session
+      return if session.present?
+      user_session[:idv] = {}
+    end
+
+    def session
+      user_session.fetch(:idv, {})
+    end
+
+    def valid_user_bundle
+      payload, headers = JWT.decode(
+        user_bundle,
+        public_key,
+        true,
+        algorithm: 'RS256',
+      )
+      @user_bundle_payload = payload
+      @user_bundle_headers = headers
+    rescue JWT::DecodeError => err
+      errors.add(:jwt, "decode error: #{err.message}", type: :invalid)
+    rescue JWT::ExpiredSignature => err
+      errors.add(:jwt, "expired signature: #{err.message}", type: :invalid)
+    end
+
+    def valid_user
+      return if current_user
+      errors.add(:user, 'user not found', type: :invalid)
+    end
+
+    def valid_password
+      return if current_user&.valid_password?(user_password)
+      errors.add(:password, 'invalid password', type: :invalid)
+    end
+
+    def form_valid?
+      @form_valid
+    end
+
+    def extra_attributes
+      if current_user.present?
+        @extra_attributes ||= {
+          personal_key: personal_key,
+          profile_pending: current_user.pending_profile?,
+          user_uuid: current_user.uuid,
+        }
+      else
+        @extra_attributes = {}
+      end
+    end
+
+    def personal_key
+      @personal_key ||= profile&.personal_key || profile&.encrypt_recovery_pii(pii)
+    end
+
+    def public_key
+      OpenSSL::PKey::RSA.new(Base64.strict_decode64(IdentityConfig.store.idv_public_key))
+    end
+
+    alias_method :jwt, :user_bundle
+    alias_method :user, :current_user
+    alias_method :password, :user_password
```

Comment:
could we just rename these methods? I feel like aliasing private methods only seems like more indirection for little gain

Reply:
Yes! done in b2ecfe1